packages/bpk-component-map/src/BpkMap.js

```diff
@@ -0,0 +1,192 @@
+
+const mapUrl = 'https://maps.googleapis.com/maps/api/js';
+const getClassName = cssModules(STYLES);
+
+type ZoomChangedCallback = ?(zoom: number) => mixed;
+
+type DragEndCallback = ?(bounds: Object, center: Object) => mixed;
+
+type Props = {
+  apiKey: string,
+  children: ?Node,
+  className: ?string,
+  language: string,
+  region: string,
+  width: string,
+  height: string,
+  zoom: number,
+  centerLatitude: ?number,
+  centerLongitude: ?number,
+  boundSouth: ?number,
+  boundWest: ?number,
+  boundNorth: ?number,
+  boundEast: ?number,
+  zoomEnabled: boolean,
+  dragEnabled: boolean,
+  onZoom: ZoomChangedCallback,
+  onDrag: DragEndCallback,
+};
+
+class BpkMap extends Component<Props> {
+  googleMap: any;
+
+  static propTypes = {
+    apiKey: PropTypes.string.isRequired,
+    children: PropTypes.node,
+    className: PropTypes.string,
+    language: PropTypes.string,
+    region: PropTypes.string,
+    width: PropTypes.string,
+    height: PropTypes.string,
+    zoom: PropTypes.number,
+    centerLatitude: PropTypes.number,
+    centerLongitude: PropTypes.number,
+    boundSouth: PropTypes.number,
+    boundWest: PropTypes.number,
+    boundNorth: PropTypes.number,
+    boundEast: PropTypes.number,
+    zoomEnabled: PropTypes.bool,
+    dragEnabled: PropTypes.bool,
+    onZoom: PropTypes.func,
+    onDrag: PropTypes.func,
+  };
+
+  static defaultProps = {
+    children: null,
+    className: null,
+    language: '',
+    region: '',
+    width: '100%',
+    height: '100%',
+    zoom: 15,
+    boundSouth: null,
+    boundWest: null,
+    boundNorth: null,
+    boundEast: null,
+    centerLatitude: null,
+    centerLongitude: null,
+    zoomEnabled: true,
+    dragEnabled: true,
+    onZoom: null,
+    onDrag: null,
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.googleMap = null;
+  }
+
+  handleMapLoad = (map: ?Object) => {
+    this.googleMap = map;
+    const { boundSouth, boundWest, boundNorth, boundEast } = this.props;
+    if (map && boundSouth && boundWest && boundNorth && boundEast) {
+      map.fitBounds({
+        south: boundSouth,
+        west: boundWest,
+        north: boundNorth,
+        east: boundEast,
+      });
+    }
+  };
+
+  handleZoomChanged = (callback: ZoomChangedCallback) => {
+    if (callback) {
+      const zoomLevel = this.googleMap.getZoom();
+      callback(zoomLevel);
+    }
+  };
+
+  handleDragEnd = (callback: DragEndCallback) => {
+    if (callback) {
+      const bounds = this.googleMap.getBounds();
+      const center = this.googleMap.getCenter();
+      callback(bounds, { lat: center.lat(), lng: center.lng() });
+    }
+  };
+
+  render() {
+    const {
+      apiKey,
+      className,
+      centerLatitude,
+      centerLongitude,
+      children,
+      width,
+      height,
+      zoom,
+      language,
+      region,
+      zoomEnabled,
+      dragEnabled,
+      onZoom,
+      onDrag,
+      ...rest
+    } = this.props;
+    const InnerMap = withScriptjs(
```

Comment:
We shouldn't be doing this here, it creates new functions everytime which breaks React's reconciling